libraries/rush-lib/src/logic/operations/ShellOperationRunner.ts

```diff
@@ -420,7 +450,7 @@ export class ShellOperationRunner implements IOperationRunner {
 
         const [, cacheWriteSuccess] = await Promise.all([writeProjectStatePromise, setCacheEntryPromise]);
 
-        if (buildCacheTerminalProvider.hasErrors) {
+        if (terminalProvider.hasErrors) {
```

Comment:
Pretty sure this was incorrectly changed here: https://github.com/microsoft/rushstack/pull/4130/files#diff-f19d02b6f5da5124f2fe15eba7801b17015e12f2d44fa9b78b8fd250abce2a17R423